### DISCUSSION TITLE
@formily/react-reactive 中的 observer 传递 ref 的方式调整

### DISCUSSION THREAD

**Discussion by liuweiGL** [alibaba/formily#1404] [Ideas]
## 问题

现在 `observer` 方法接受一个 `React.ForwardRefRenderFunction<any>` 组件才能正常转发  ref，导致自定义的组件无法使用 `React.ComponentProps<T>` 推断出组件的 props 类型。

## 期望

`observer` 能否接受一个 `ForwardRefExoticComponent` 组件进行 ref 的转发？这样的话跟 `connect` 需要的组件类型也保持一致了。

**Comment by janryWang** [comment]
可以的，整个PR吧

**Reply by liuweiGL** [reply]
恩，晚点。

**Reply by janryWang** [reply]
炜兄最近很高产，点个赞👍
